docs/internals/optimizer.rst

```diff
@@ -411,6 +411,8 @@ its visibility and it is impossible to reference variables defined in a differen
 The benefit of this stage is that function definitions can be looked up more easily
 and functions can be optimized in isolation without having to traverse the AST completely.
 
+Prerequisites: Disambiguator.
```

Comment:
Note that not all steps list disambiguator as a prerequisite even though its own description says that all of them depend on it.

BTW, not sure if listing prerequisites is all that useful since we have very few hard prerequisites and those are hard-coded anyway. Maybe it would be fine to remove this info. For now though we do list that info so I amended it for steps where it was available in the docstrings.